src/python/library/tritonclient/utils/_dlpack.py

```diff
@@ -0,0 +1,234 @@
+def pycapsule_deleter(handle: ctypes.c_void_p) -> None:
+    pycapsule: ctypes.py_object = ctypes.cast(handle, ctypes.py_object)
+    if ctypes.pythonapi.PyCapsule_IsValid(pycapsule, c_str_dltensor):
+        dl_managed_tensor = ctypes.pythonapi.PyCapsule_GetPointer(
+            pycapsule, c_str_dltensor
+        )
+        managed_tensor_deleter(dl_managed_tensor)
+        ctypes.pythonapi.PyCapsule_SetDestructor(pycapsule, None)
+
+def triton_to_dlpack_dtype(dtype):
+    if dtype == "BOOL":
+        type_code = DLDataTypeCode.kDLBool
+        bits = 1
+    elif dtype == "INT8":
+        type_code = DLDataTypeCode.kDLInt
+        bits = 8
+    elif dtype == "INT16":
+        type_code = DLDataTypeCode.kDLInt
+        bits = 16
+    elif dtype == "INT32":
+        type_code = DLDataTypeCode.kDLInt
+        bits = 32
+    elif dtype == "INT64":
+        type_code = DLDataTypeCode.kDLInt
+        bits = 64
+    elif dtype == "UINT8":
+        type_code = DLDataTypeCode.kDLUInt
+        bits = 8
+    elif dtype == "UINT16":
+        type_code = DLDataTypeCode.kDLUInt
+        bits = 16
+    elif dtype == "UINT32":
+        type_code = DLDataTypeCode.kDLUInt
+        bits = 32
+    elif dtype == "UINT64":
+        type_code = DLDataTypeCode.kDLUInt
+        bits = 64
+    elif dtype == "FP16":
+        type_code = DLDataTypeCode.kDLFloat
+        bits = 16
+    elif dtype == "FP32":
+        type_code = DLDataTypeCode.kDLFloat
+        bits = 32
+    elif dtype == "FP64":
+        type_code = DLDataTypeCode.kDLFloat
+        bits = 64
+    elif dtype == "BF16":
+        type_code = DLDataTypeCode.kDLBfloat
+        bits = 16
+    elif dtype == "BYTES":
+        _raise_error("DLPack currently doesn't suppose BYTES type")
+    else:
+        _raise_error("Can not covert unknown data type '{}' to DLPack data type".format(dtype))
+    return DLDataType(type_code, bits, 1)
+
+def is_support_device(device: DLDevice):
+    return device[0] in [DLDeviceType.kDLCPU, DLDeviceType.kDLCUDA, DLDeviceType.kDLCUDAHost]
+
+def is_contiguous_data(ndim: ctypes.c_int, shape: ctypes.POINTER(ctypes.c_int64), stride: ctypes.POINTER(ctypes.c_int64)):
+    # If 'stride' doesn't capture valid value
+    if (stride is None) or (not bool(stride)):
+        return True
+    calculated_stride = 1
+    # iterate stride in reverse order [ndim-1, -1)
+    for i in reversed(range(ndim)):
+        if stride[i] != calculated_stride:
+            return False
+        calculated_stride *= shape[i]
+    return True
+
+def get_byte_size(dtype: DLDataType, ndim: ctypes.c_int, shape: ctypes.POINTER(ctypes.c_int64)):
+    element_byte_size = dtype.bits * dtype.lanes // 8 # Assume 8 bits in a byte
+    for i in range(ndim):
+        element_byte_size *= shape[i]
+    return element_byte_size
+
+def get_dlpack_capsule(dlpack_obj):
+    # Extract PyCapsule of the DLPack object
+    if hasattr(dlpack_obj, '__dlpack__'):
+        device = dlpack_obj.__dlpack_device__()
```

Comment:
nit: I would also check that `__dlpack_device__` is available

Reply:
Isn't `__dlpack_device__` required?

Reply:
I personally would check, but up to you. By standard, `__dlpack__` and `__dlpack_device__` both required at the same time.